NDTensors/src/UnallocatedArrays/src/unallocatedfill.jl

```diff
@@ -0,0 +1,14 @@
+## TODO All constructors not fully implemented but so far it matches the 
+## constructors found in `FillArrays`. Need to fix io
+struct UnallocatedFill{ElT,N,Axes,Alloc<:AbstractArray} <:
+       FillArrays.AbstractFill{ElT,N,Axes}
+  f::FillArrays.Fill{ElT,N,Axes}
+
+  function UnallocatedFill{ElT,N,Axes,Alloc}(x::ElT, inds::Tuple) where {ElT,N,Axes,Alloc}
+    f = FillArrays.Fill(x, inds)
+    ax = typeof(FillArrays.axes(f))
+    return new{ElT,N,ax,Alloc}(f)
```

Comment:
You should use `Axes` here, what you wrote could potentially change the `Axes` type.

Reply:
Also, if we go with my suggested design of `set_alloctype` this constructor may not be necessary.

I'm a bit worried we're going to have too many constructors/converters whereas if we have some generic code concepts we might be able to avoid that.

Reply:
So right now I have removed all constructors and this is most likely fine, we can still construct a UnallocatedFill/Zero with
```
F = Fill(n, (i,j,k))
Alloc = Array{eltype(F), ndims(F)}
UnallocatedFill{eltype(F), ndims(F), typeof(axes(F)), Alloc}(F)
```
And I wil lbe making the `set_alloctype` based constructor that you suggested above